ServiceHandler: Implements methods to change proxys dependencies

CarlosCraveiro opened the pull request:
<!--PS: 🇧🇷/🇬🇧 It's ok to write your PR in Portuguese 😃-->

## What type of PR is this?
- [x] ✨ Feature
- [ ] 🐞 Bug Correction
- [ ] 📝 Documentation Update
- [ ] 🚩 Other

# Changes:
<!-- Describe the main changes and the **expected behavior** (if applicable) -->
Esse pull request foi motivado pela necessidade de implementação dos testes unitátios para as "classes services". 

Durante a tentativa de implementar os testes unitários, senti uma enorme necessidade de fazer mudanças nas dependencias dos proxys contidos no interior do ServiceHandler, mas eu não havia implementado nenhuma maneira de fazer isso. Essa necessidade não apareceu durante a elaboração do ServiceHandler, pois acreditava que que essa dependencias seriam trocadas manualmente por uma outra classe agregada que teria o privilégio de amizade com as classes service. Porém, almejando testes unitários robustos percebi que não haveria mal algum em fornecer métodos públicos no ServiceHandler para fazer essas alterações, visto que estes poderão  ser utilizados também por essa classe que está por vir, eles permitirão testes unitários mais completos e noo pior dos casos, o "user" final do framework não terá acesso à classe ServiceHandler dirertamente pela classe daemon, logo não haverá "riscos".
(Service Handler e a saga continua...)

# Testing
<!--(if applicable) Tell us how to replicate the expected behavior.-->
Esse pull request compila e funciona assim como a versão anterior.

CarlosCraveiro committed:
ServiceHandler: Implements methods to change proxys dependencies

New methods of the ServiceHandler class were required to allow unit
testing of the collection of service classes.

This methods simply provides an API to change the dependencies states
of each proxy stored inside the ServiceHandler.

mairacanal commented on src/classes/service-handler/service-handler.cpp:
```diff
@@ -28,7 +28,17 @@ void ServiceHandler::buildServiceProxy(RoutineService& userService)
+        throw std::invalid_argument(errorMessage);
+    }
+
     return requiredState;
```
```suggestion
    try {
        return m_serviceMap.at(serviceId)->reportState();
    } catch (const std::out_of_range) {
        std::string errorMessage = "Get dependency state request from a not built service! Unknown Service = " + serviceId;

        throw std::invalid_argument(errorMessage);
    }
```

mairacanal commented on src/classes/service-handler/service-handler.cpp:
```diff
@@ -43,6 +53,41 @@ auto ServiceHandler::getAllProxyState() -> nlohmann::json
+    try {
+        m_serviceMap.at(serviceId)->m_proxyConfigs.changeDep(dependencyId, newState);
+    } catch (const std::out_of_range) {
+        std::string errorMessage = "Change dependency state request from a not built service! Unknown Service = ";
+        errorMessage.append(serviceId);
```
```suggestion
        std::string errorMessage = "Change dependency state request from a not built service! Unknown Service = " + serviceId;
```

mairacanal commented on src/classes/service-handler/service-handler.cpp:
```diff
@@ -43,6 +53,41 @@ auto ServiceHandler::getAllProxyState() -> nlohmann::json
+    try {
+        m_serviceMap.at(serviceId)->autoUpdate();
+    } catch (const std::out_of_range) {
+        std::string errorMessage = "Update request from a not built service! Unknown Service = ";
+        errorMessage.append(serviceId);
```
```suggestion
        std::string errorMessage = "Update request from a not built service! Unknown Service = " + serviceId;
```

mairacanal reviewed (changes requested):
Código ótimo, apenas alguns pequenos comentários relativos a sintaxe.

Grillo-0 reviewed (dismissed):
PR muito bom, parabéns 🎉

CarlosCraveiro committed:
ServiceHandler: Improves syntax

Adopts the overload of the "+" operator to concatenate two strings instead of using ".append()".

Co-authored-by: Maíra Canal <61601791+mairacanal@users.noreply.github.com>

Math-42 reviewed (approved):
Código ótimo, parabéns!
